src/test/OpenEXRTest/testHeader.cpp

```diff
@@ -0,0 +1,109 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) Contributors to the OpenEXR Project.
+//
+
+#ifdef NDEBUG
+#    undef NDEBUG
+#endif
+
+#include <ImfBoxAttribute.h>
+#include <ImfHeader.h>
+
+#include <exception>
+#include <iostream>
+#include <string>
+
+#include <assert.h>
+
+using namespace IEX_NAMESPACE;
+using namespace IMATH_NAMESPACE;
+using namespace OPENEXR_IMF_NAMESPACE;
+using namespace std;
+
+template<typename Header>
+struct Test
+{
+    void testFind(const string& name)
+    {
+        Header header;
+        auto iterator = header.find(name);
+        assert(iterator != header.end());
+    }
+
+    void testSubscript(const string& name)
+    {
+        Header header;
+        auto& comparand = header.find("displayWindow").attribute();
+        auto& attribute = header[name];
+        assert(&attribute == &comparand);
+    }
+
+    void testIterators(const string& name)
+    {
+        Header header;
+
+        auto& comparand = header.find("displayWindow").attribute();
+
+        for (auto iterator = header.begin(); iterator != header.end(); ++iterator)
+        {
+            if (iterator.name() == name)
+            {
+                assert(&iterator.attribute() == &comparand);
+                return;
+            }
+        }
+
+        assert(false);
+    }
+};
+
+void testEraseKnownAttribute(const string& name)
+{
+    Header header;
+    header.erase(name);
+    assert(header.find(name) != header.end());
```

Comment:
Shouldn't that be `header.find(name) == header.end()` since it's been erased?

Reply:
Oh yes, I'm sorry, I made it != it to ensure that failing tests were being reported for me. For a while at the beginning the assert statements weren't being evaluated and I didn't realise. That was fixed when I realised what the `#ifdef NDEBUG` section at the top of the file was doing, but I left this to make sure things kept reporting correctly and then forgot to correct it before committing.

Reply:
Thanks, just push a new commit when you're ready.

The CLA should be in place now, too.  Try clicking through the CLA link, with Lucasfilm as the company.

Reply:
Thanks, that's all sorted now I think!